src/main/java/org/apache/commons/lang3/RandomStringUtils.java

```diff
@@ -330,8 +329,8 @@ public static String random(int count, int start, int end, final boolean letters
         // Ideally the cache size depends on multiple factor, including the cost of generating x bytes
         // of randomness as well as the probability of rejection. It is however not easy to know
         // those values programmatically for the general case.
-        // To prevent overflow restrict the cache size to 60M entries (
-        final int cacheSize = Math.min ((count * gapBits + 3) / 5 + 10, MAX_CACHE_SIZE);
+        // For huge strings the padding required can cause an overflow
+        final int cacheSize = (count * gapBits + 3) / 5 + 10 > 0 ? (count * gapBits + 3) / 5 + 10 : Integer.MAX_VALUE;
```

Comment:
```suggestion
        final int cacheSize = count * gapBits + 3 > 0 ? (count * gapBits + 3) / 5 + 10 : Integer.MAX_VALUE;
```

Adding `10` can transform a negative number into a small positive one.

Reply:
Fixed in https://github.com/apache/commons-lang/pull/1379/commits/659e14a74eea438ba0a58edb9c11886391ff3517